airflow/executors/base_executor.py

```diff
@@ -42,7 +43,124 @@
 QueuedTaskInstanceType = Tuple[CommandType, int, Optional[str], SimpleTaskInstance]
 
 
-class BaseExecutor(LoggingMixin):
+class BaseExecutorProtocol(LoggingMixin):
+    """
+    Base Protocol implemented by all executors including multiple executors.
+    """
+
+    def __init__(self):
+        super().__init__()
```

Comment:
If this has code it's not a "Protocol"

Reply:
Yeah. There are no real interfacces/protocols in Python. It's a bit of a pity but also adds more flexibility.